rules/enforce-eslint-mandatory-prop.js

```diff
@@ -0,0 +1,46 @@
+const hasProp = require("jsx-ast-utils/hasProp");
+
+module.exports = {
+  /** @type {import('eslint').Rule.RuleMetaData} */
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Enforce passing a mandatory prop to a component",
+    },
+    schema: [
+      {
+        type: "object",
+        additionalProperties: {
+          type: "object",
+          additionalProperties: {
+            type: "object",
+            additionalProperties: {
+              type: "boolean",
+            },
+          },
+        },
+      },
```

Comment:
Do you need all that? I think this defines the schema of the eslint config and it doesn't seem to be really used here

Reply:
I thought the same, but saw the comment here and added it so we can upgrade eslint version in the future
```ts
        /**
         * Specifies the [options](https://eslint.org/docs/latest/developer-guide/working-with-rules#options-schemas)
         * so ESLint can prevent invalid [rule configurations](https://eslint.org/docs/latest/user-guide/configuring/rules#configuring-rules).
         * TODO: schema is potentially planned to be no longer be optional in v9 (https://github.com/eslint/rfcs/blob/main/designs/2021-schema-object-rules/README.md)
         */
        schema?: JSONSchema4 | JSONSchema4[] | undefined;
```

Reply:
Yes, but that's not defining the right structure here 😅

Reply:
I think it is - it's [{config: {'Component': {prop: true}}}] so 3 objects, then a key-value with booleans. I think it's beneficial to have it to protect against wrong configuration that would break this rule:

![image](https://github.com/plentific/eslint-plugin-plentific/assets/25098043/9ccf08f4-bdd3-433a-acf9-10a12e3df4cb)

The config key may be misleading, let me remove it

Reply:
Oh, sorry, my bad. Looks good now 👍

Reply:
Oh, sorry, my bad. Looks good now 👍